rocketpy/rocket/components.py

```diff
@@ -186,3 +186,18 @@ def sort_by_position(self, reverse=False):
         None
         """
         self._components.sort(key=lambda x: x.position, reverse=reverse)
+
+    def to_dict(self):
+        """Return a dictionary representation of the components.
```

Comment:
You`re gonna have to make a lot of `to_dict` methods. My recommendation is to avoid documenting all of them. 
It's still experimental feature, so let's be "lazy" this time.